opm/autodiff/NewtonIterationBlackoilCPR.cpp

```diff
@@ -188,11 +189,14 @@ namespace Opm
             const ParallelISTLInformation& info =
                 boost::any_cast<const ParallelISTLInformation&>( parallelInformation_);
             Comm istlComm(info.communicator());
-            info.copyValuesTo(istlComm.indexSet(), istlComm.remoteIndices());
+            Comm istlAeComm(info.communicator());
+            info.copyValuesTo(istlAeComm.indexSet(), istlAeComm.remoteIndices());
+            info.copyValuesTo(istlComm.indexSet(), istlComm.remoteIndices(),
+                              istlAe.N(), istlA.N()/istlAe.N());
```

Comment:
I have some trouble understanding what really happens in the preceding four lines. Could you try to explain?